pkg/k8s/watchers/subscriber/node.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriber
+
+import (
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+)
+
+// NodeHandler is implemented by event handlers responding to K8s Node events.
+type NodeHandler interface {
```

Comment:
@christarazi 

While this is a bit subjective, my initial read-thru was confused by this terminology of "NodeHandler". In context with the package name we get "subscriber.NodeHandler". Handler to me makes me think HTTP handler not a "subscriber". 

I'm not too hell-bent on this but if you are willing to consider alternative names here are a few that would have helped me grok this easier. 

"subscriber.NodeEvents"
"subscriber.Node"
"subscriber.NodeSubscriber" 

With those names it becomes apparent that "okay, if I want to write a subscriber which listens for Node events, I want to implement a {one of the above}". 

again, subjective, but let me know if you're open to thinking thru the naming.

Reply:
Yes, that's a good suggestion. I would definitely be open to improving the naming. In my head, I was kinda interchanging "handler" and "subscriber", so we should just go with the latter everywhere.

Reply:
Which of the above naming suggestion do you think is best? Feel free to suggest others as well.

Reply:
I personally like "subscriber.Node" the best because "subscriber.NodeSubscriber" stutters (redundant "subscriber") and the first one I don't see the value of saying "Events". The middle one is sufficient.

Reply:
Agreed